File: app/src/main/java/com/star_zero/example/androidmvvm/presentation/shared/view/BaseActivity.java
package com.star_zero.example.androidmvvm.presentation.shared.view;

import android.support.v7.app.AppCompatActivity;

import com.star_zero.example.androidmvvm.App;
import com.star_zero.example.androidmvvm.di.AppComponent;

public abstract class BaseActivity extends AppCompatActivity {

    protected AppComponent getAppComponent() {
        return ((App) getApplication()).getAppComponent();
    }
}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/presentation/shared/view/BaseActivity.kt
package com.star_zero.example.androidmvvm.presentation.shared.view

import android.arch.lifecycle.LifecycleRegistry
import android.arch.lifecycle.LifecycleRegistryOwner
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity(), LifecycleRegistryOwner {
    val registry: LifecycleRegistry by lazy {
        LifecycleRegistry(this)
    }

    override fun getLifecycle(): LifecycleRegistry {
        return registry
    }

}

